File: examples/rest-api/nodejs/src/functions/UpdateProduct.ts
import ProductService from "../services/ProductService";
import {
	APIGatewayProxyEvent,
	APIGatewayProxyResult,
	Context
} from "aws-lambda";
import Product from "../models/Product";

var service = new ProductService(process.env.PRODUCTS_TABLE_NAME);
export const handler = async (
	event: APIGatewayProxyEvent,
	context: Context
): Promise<APIGatewayProxyResult> => {
	context.callbackWaitsForEmptyEventLoop = false;

	if (!event.body) {
		return {
			statusCode: 400,
			body: "Bad Request, body cannot be empty"
		};
	} else {
		let product: Product = JSON.parse(event.body!);
		try {
			var item = await service.updateItem(product);
			return { statusCode: 200, body: JSON.stringify(item) };
		} catch (e) {
			return {
				statusCode: 500,
				body: "Server failure, unable to retrieve item"
			};
		}
	}
};

File: examples/rest-api/csharp-lambda-layer/functions/UpdateProduct.cs
using Amazon.Lambda.Core;
using System;
using Amazon.Lambda.APIGatewayEvents;
using System.Threading.Tasks;
using RestApiServices;
using Newtonsoft.Json;
using Models;
using Amazon.DynamoDBv2.DocumentModel;

namespace RestApi
{
    public class UpdateProduct
    {
        ProductService service = new ProductService(Environment.GetEnvironmentVariable("PRODUCTS_TABLE_NAME"));

        [LambdaSerializer(typeof(Amazon.Lambda.Serialization.Json.JsonSerializer))]
        public async Task<APIGatewayProxyResponse> Handler(APIGatewayProxyRequest request, ILambdaContext context)
        {
            var body = request.Body;
            Product product = JsonConvert.DeserializeObject<Product>(body);

            if (string.IsNullOrEmpty(product.sku))
            {
                return new APIGatewayProxyResponse
                {
                    StatusCode = 400,
                    Body = "sku cannot be empty"
                };
            }

            try
            {
                Document item = await service.UpdateItem(product);

                return new APIGatewayProxyResponse
                {
                    StatusCode = 200,
                    Body = item == null ? "" : item.ToJson()
                };
            }
            catch (AggregateException e)
            {
                Console.WriteLine(e);
                return new APIGatewayProxyResponse
                {
                    StatusCode = 500,
                    Body = ""
                };
            }
        }
    }
}
